src/Ocelot/Configuration/Creator/TimeoutCreator.cs

```diff
@@ -0,0 +1,14 @@
+using Ocelot.Configuration.File;
+
+namespace Ocelot.Configuration.Creator
+{
+    public class TimeoutCreator : ITimeoutCreator
+    {
+        public int Create(FileRoute fileRoute, FileGlobalConfiguration globalConfiguration)
+        {
+            return fileRoute.Timeout > 0 
+                ? fileRoute.Timeout 
+                : (globalConfiguration.RequestTimeoutSeconds ?? 0) * 1000;
```

Comment:
After the changes, the `globalConfiguration.RequestTimeoutSeconds` will never be null. Maybe it would be wise to use a unique naming convention everywhere.

Reply:
We haven't decided on nullable types yet!
Seems we have the conflict of two design visions... Could we discuss this privately in Slack please?
We need a consensus here definitely! 

**Second**, Regarding creator class... It will be added to DI. Do you believe this micro service class should be in DI and overridden in theory? What's your vision of allowing this for end developers?
I'm just curious about possible user scenarios of overriding this creator?
**Pay attention we have end consumer class which is `TimeoutDelegatingHandler`❗** 

**Third**, both of you guys forgot about Polly QoS value 🤣 
I think we have 3 or even 4 values to merge somewhere:
- current `Timeout` at Route-level
- global timeout seconds for all routes
- Polly QoS `TimeoutValue` of QoS options
- default 90 seconds value

💥 Bingo!